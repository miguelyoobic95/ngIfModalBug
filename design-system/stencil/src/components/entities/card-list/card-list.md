---
name: Card List
category: Entities Components
---

```card-list.html
        <yoo-card-list heading="Card List" class="accent" img-src="https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg" id="list"></yoo-card-list>
```
```card-list.js
    var listMode = document.querySelector('#list');
    var tags = [{text: 'Finished', icon:'yo-flag'},{text: 'Fire Account', icon:'yo-fire'}];
    listMode.tags = tags;
    listMode.topLeftBadge = "PDF";
    listMode.BottomRightBadge = "PDF";
    listMode.subheadings = ["subtitle 1", "subtitle 2"];
    listMode.badges = [{text: 'Finished', icon:'yo-home'}, {text:'In-Progress'}];
```


## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`text`|string|   |the text inside the label|

## CSS

|Type|Name|Description|
|---|---|---|


## Events
|Attr|Description|
|---|---|---|