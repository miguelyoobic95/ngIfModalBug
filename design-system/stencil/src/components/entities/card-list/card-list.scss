@mixin badge-position() {
    yoo-form-checkbox {
        position: absolute;
        top: 0.5rem;
        left: 0.5rem;
    }
    .bottom-left,
    .bottom-right,
    .top-left,
    .top-right {
        z-index: 1;
        position: absolute;
    }
    .bottom-left {
        bottom: 0.5rem;
        left: 0.5rem;
    }
    .bottom-right {
        bottom: 0.5rem;
        right: 0.5rem;
    }
    .top-left {
        top: 0.5rem;
        left: 0.5rem;
        &.active {
            top: 2rem;
        }
    }
    .top-right {
        top: -0.2rem;
        right: 0.1rem;
    }
}

:host {
    display: block;
    .outer-container {
        width: 100%;
        border-bottom: 1px solid cssvar(dark-10); //border-radius: 2px;
        //box-shadow: 0px 0px 3px 1px cssvar(dark-10); //padding-bottom: 0.5rem;
        .image-container {
            position: relative;
            margin-top: 0.2rem;
            margin-bottom: 0.2rem;
            @include badge-position();
            yoo-avatar.main,
            .multiple-avatar-container {
                margin-left: 0.8rem;
                margin-right: 0.3rem;
                align-self: center;
            }
            .multiple-avatar-container {
                .top-avatars {
                    yoo-avatar {
                        transform: translateY(3px);
                    }
                }
                .avatar-hidden {
                    height: 22px;
                    width: 22px;
                    background-color: cssvar(stable);
                    color: cssvar(light);
                    font-size: 0.6rem;
                    border-radius: 50%;
                    display: flex;
                    justify-content: center;
                    align-items: center;
                }
            }
        }
        .content-container {
            background: cssvar(light);
            width: 100%;
            justify-content: space-between;
            padding: 0.5rem;
            display: flex;
            flex-direction: column;
            .heading-container {
                font-size: 1rem;
                .card-heading {
                    flex: 1;
                }
            }
            .subheading-container {
                font-size: 0.9rem;
                color: cssvar(text-color);
                margin-top: 0.2rem;
            }
            .badges-container {
                align-items: center;
                yoo-badge,
                .card-icon {
                    margin-right: 0.625rem;
                }
                .hashtag {
                    color: cssvar(text-color);
                    font-size: 0.6875rem;
                    margin-right: 0.4375rem;
                    transform: translateY(0.125rem);
                }
                .card-icon {
                    display: flex;
                    align-items: center;
                    i {
                        margin-right: 0.4375rem;
                        font-size: 1.2rem;
                    }
                }
            }
            .avatar-container {
                margin-top: 0.5rem;
                justify-content: flex-end;
                padding-right: 0.5rem;
                yoo-avatar {
                    margin-right: 0.15rem;
                    margin-left: 0.15rem;
                }
            }
            .action-button-container {
                justify-content: flex-end;
            }
            .slot-container {
                font-size: 0.9rem;
                color: cssvar(text-color);
                justify-content: space-between;
                .inner-container {
                    justify-content: flex-end;
                    align-items: center;
                }
            }
            .top-container {
                .heading-container {
                    .card-heading {
                        font-size: 1.0625rem;
                    }
                    .date-card {
                        color: cssvar(text-color);
                        font-size: 0.9rem;
                    }
                }
            }
        }
        &.active {
            box-shadow: 0px 0px 3px 1px cssvar(accent);
            border: 1px solid cssvar(accent);
        }
    }
}

:host(.feedsComments) {
    .outer-container {
        padding-top: 0.625rem;
        .content-container {
            .heading-container {
                font-size: 0.9375rem;
            }
            .subheading-container {
                font-size: 0.9375rem;
                color: cssvar(dark);
            }
            .top-container {
                .badges-container {
                    margin-top: 0.5rem;
                    font-size: 0.75rem;
                    .bottom-action {
                        color: cssvar(text-color);
                        margin-right: 1.25rem;
                    }
                }
            }
        }
        .image-container {
            yoo-avatar.main {
                align-self: normal;
            }
        }
    }
}

:host(.active) {
    .outer-container {
        box-shadow: 0px 0px 3px 1px cssvar(accent);
        border: 1px solid cssvar(accent);
    }
}

:host(.hoverable) {
    &:hover {
        .outer-container {
            box-shadow: 0px 0px 3px 1px cssvar(accent);
            border: 1px solid cssvar(accent);
        }
    }
}

:host(.filesFolders),
:host(.files) {
    .outer-container {
        .image-container {
            yoo-avatar.main {
                margin-left: 0;
            }
        }
        .content-container {
            padding: 0.3rem;
            .subheading-container {
                font-size: 0.625rem;
                margin-top: 0;
            }
        }
    }
}