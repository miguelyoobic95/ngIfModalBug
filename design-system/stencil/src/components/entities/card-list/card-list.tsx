import { Component, Element, Prop, Event, State, EventEmitter } from '@stencil/core';
//import { resizeObserve, debounce } from '../../../utils/helpers'; //intersectionObserve
//import { setAnimation, animations } from '../../../utils/anim';
import { execHandlerAndStopEvent } from '../../../utils/helpers'; //intersectionObserve, resizeObserve, debounce
import { IUser, ICardListEntry } from '@shared/interfaces';

const MAX_AVATAR_DISPLAY: number = 4;

@Component({
    tag: 'yoo-card-list',
    styleUrl: 'card-list.scss',
    scoped: true
})
export class YooCardListComponent {

    @Prop() entry: ICardListEntry;

    @Prop() avatarImgs: string[] = [];
    @Prop() isActivable: boolean = false;
    @Prop() actionButtonTitle: string;
    @Prop() animationName: string;
    @Prop() hasMenu: boolean = false;

    @Event() active: EventEmitter<boolean>;
    @Event() actionPress: EventEmitter<boolean>;

    @State() isActive: boolean = false;
    @State() horizontal: boolean;
    @State() imageWidth: number = 340;
    @State() imageHeight: number = 160;

     @Element() host: HTMLStencilElement;

    //private resizeObserver: ResizeObserver;

    componentWillLoad() {
        // if (this.entry && !this.entry.imgSrc) {
        //     this.entry.imgSrc = '../../assets/empty-states/empty.svg';
        // }
    }

    componentDidLoad() {
        // let image = this.host.querySelector('.image');

        // this.resizeObserver = resizeObserve(image, (target, width, height, left, top, entry) => {
        //     debounce(this.onImageResize.bind(this), 1000)(target, width, height, left, top, entry);
        // });
    }

    componentDidUnload() {
        // if (this.resizeObserver) {
        //     this.resizeObserver.disconnect();
        // }
    }

    onImageResize(target, width, height, left, top, entry) {
        // this.imageWidth = width;
        // this.imageHeight = height;
    }

    onCheckboxToggled(event): void {
        event.detail === 'checked' ? this.isActive = true : this.isActive = false;
        this.active.emit(this.isActive);
    }

    onActionButtonClick(): void {
        this.actionPress.emit(true);
    }

    renderCardImage(): JSX.Element {
        let orderUser = this.entry.users ? this.entry.users.sort((a: IUser, b: IUser) => {
            return a.imageData && b.imageData ? 0 : !b.imageData ? -1 : 1;
        }) : [];
        return (this.entry && (!this.entry.users || !this.entry.users.length) ?
            <yoo-avatar class={'main ' + (this.entry && this.entry.avatarSize ? this.entry.avatarSize : 'list-small')} img-src={this.entry.imgSrc} icon={this.entry.icon} icon-text={this.entry.iconText}></yoo-avatar>
            : this.entry && this.entry.users && this.entry.users.length === 1 ?
                <yoo-avatar class={'main ' + (this.entry && this.entry.avatarSize ? this.entry.avatarSize : 'list-small')} img-src={this.entry.imgSrc} user={this.entry.users[0]}></yoo-avatar>
                : this.entry && this.entry.users ?
                    <div class="multiple-avatar-container" attr-layout="column">
                        <div attr-layout="row" class="top-avatars">
                            {orderUser.slice(0, 2).map((user, index) => {
                                return <yoo-avatar class="xsmall" user={user}></yoo-avatar>;
                            })}
                        </div>
                        <div attr-layout="row">
                            {orderUser.slice(2, 3).map((user, index) => {
                                return <yoo-avatar class="xsmall" user={user}></yoo-avatar>;
                            })}
                            {this.entry.users.length > 3 ?
                                <div class="avatar-hidden" attr-layout="row">
                                    <span>+{this.entry.users.length - 3}</span>
                                </div>
                                : null}
                        </div>
                    </div>
                    : null);
    }

    renderBottomContent(): JSX.Element {
        return (
            [
                this.avatarImgs && this.avatarImgs.length > 0 ?
                    <div class="avatar-container" attr-layout="row">
                        {this.avatarImgs.map((avatarSrc, index) => {
                            if (index < MAX_AVATAR_DISPLAY) {
                                return (<yoo-avatar class="xsmall" img-src={avatarSrc}></yoo-avatar>);
                            }
                        })}
                    </div> : null,
                this.actionButtonTitle ?
                    <div class="action-button-container" attr-layout="row">
                        <yoo-button text={this.actionButtonTitle} class={this.host.className + ' squared'} onClick={() => this.onActionButtonClick()}></yoo-button>
                    </div> : null]
        );
    }

    renderImageContainerContent(): JSX.Element {
        return (
            [this.entry.topLeftBadge ? (
                <div class={'top-left' + (this.isActivable ? ' active' : '')}>
                    <yoo-badge text={this.entry.topLeftBadge} class="notification-medium danger">
                    </yoo-badge></div>) : null,
            this.entry.topRightBadge ? (
                <div class="top-right">
                    <yoo-badge text={this.entry.topRightBadge} class="notification-medium danger">
                    </yoo-badge></div>) : null,
            this.entry.bottomLeftBadge ? (
                <div class="bottom-left">
                    <yoo-badge text={this.entry.bottomLeftBadge} class="notification-medium danger">
                    </yoo-badge></div>) : null,
            this.entry.bottomRightBadge ? (
                <div class="bottom-right">
                    <yoo-badge text={this.entry.bottomRightBadge} class="notification-medium danger">
                    </yoo-badge></div>) : null,
            this.renderCardImage(),
            this.isActivable ?
                <yoo-form-checkbox class={this.host.className} onCheckboxToggled={(event) => this.onCheckboxToggled(event)}></yoo-form-checkbox>
                : null]
        );
    }

    renderHeadingContainerContent(): JSX.Element {
        return (
            [<div class="heading-container" attr-layout="row">
                {this.entry && this.entry.heading ? <span class="card-heading">{this.entry.heading}</span> : null}
                {this.entry && this.entry.date ? <span class="date-card">{this.entry.date}</span> : null}
                {this.hasMenu ?
                    <yoo-context-menu>
                        <div slot="trigger"><span class="menu-icon"><i class="yo-more-v"></i></span></div>
                        <div class="context-container">
                            <slot name="menu-slot" />
                        </div>
                    </yoo-context-menu>
                    : null}
            </div>,
            (this.entry && this.entry.topActions && this.entry.topActions.length ?
                <div>
                    {this.entry.topActions.map(a =>
                        <span onClick={(ev) => execHandlerAndStopEvent(ev, a.handler)}>{a.text}</span>
                    )}
                </div>
                 : null ),
            (this.entry && this.entry.subheadings ? this.entry.subheadings.map((item) =>
                <div class="subheading-container" innerHTML={item}></div>
            ) : null),
            (this.entry && (this.entry.badges || this.entry.tags || this.entry.icons || this.entry.bottomActions) ?
                <div class="badges-container" attr-layout="row">
                    {this.entry.badges ? this.entry.badges.map((item) =>
                        <yoo-badge class={item.cssClass ? item.cssClass : ''} icon-left={item.iconLeft} text={item.text} closable={item.closable} ></yoo-badge>)
                        : null}
                    {this.entry.tags ? this.entry.tags.map(a =>
                        <span class="hashtag" innerHTML={`#${a} `}></span>)
                        : null}
                    {this.entry.icons ? this.entry.icons.map(icon =>
                        icon && icon.icon ?
                            <span class="card-icon" attr-layout="row" onClick={(ev) => execHandlerAndStopEvent(ev, icon.handler)}>
                                <i class={icon.icon} />{' ' + icon.value ? icon.value : ''}
                            </span>
                            : null)
                        : null}
                    {this.entry && this.entry.bottomActions ?
                        <div attr-layout="row">
                            {this.entry.bottomActions.map(a =>
                                <span class="bottom-action" onClick={(ev) => execHandlerAndStopEvent(ev, a.handler)}>{a.text}</span>
                            )}
                        </div>
                    : null}
                </div>
                : null)
            ]
        );
    }

    render(): JSX.Element {
        return (
            <div class={'outer-container '} attr-layout="row">
                {this.entry && (this.entry.imgSrc || this.entry.users || this.entry.icon || this.entry.iconText)  ?
                    <div class="image-container" attr-layout="row">
                        {this.renderImageContainerContent()}
                    </div> : null}
                <div class={'content-container ' + (this.entry && this.entry.heading === undefined && this.entry.subheadings === undefined ? 'center' : '')} attr-layout="column">
                    <div class="top-container">
                        {this.renderHeadingContainerContent()}
                    </div>
                    <div class="slot-container" attr-layout="row">
                        <slot name="content-slot" />
                        <div class="inner-container" attr-layout="column">
                            {this.renderBottomContent()}
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
