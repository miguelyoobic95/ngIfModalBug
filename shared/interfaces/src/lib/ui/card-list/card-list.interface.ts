import { IUser } from '../../entities/user/user.interface';
import { IBadgeEntry } from '../badge/badge.interface';
import { IActionSheetButton } from '../action-sheet/action-sheet.interface';

export interface ICardListEntry {
    heading?: string;
    subheadings?: string[];
    imgSrc?: string;
    icon?: string;
    iconText?: string;
    avatarSize?: string;
    topLeftBadge?: string;
    topRightBadge?: string;
    bottomLeftBadge?: string;
    bottomRightBadge?: string;
    date?: string;
    badges?: IBadgeEntry[];
    users?: Array<IUser>;
    tags?: Array<string>;
    icons?: Array<{ icon: string, value?: string, handler?: () => void }>;
    actions?: Array<IActionSheetButton>;
    topActions?: Array<{ text: string, handler?: () => void }>;
    bottomActions?: Array<{ text: string, handler?: () => void }>;
}