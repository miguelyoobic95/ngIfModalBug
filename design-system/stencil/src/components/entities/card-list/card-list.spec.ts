import { TestWindow } from '@stencil/core/testing';
import { YooCardListComponent } from './card-list';

describe('CardListComponent', () => {
    it('should build', () => {
        expect(new YooCardListComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        let element, window;
        beforeEach( async () => {
            window = new TestWindow();
            element = await window.load({
                components: [YooCardListComponent],
                html: '<yoo-card-list></yoo-card-list>'
            });
        });

        it ('Should render', async () => {
            expect(element).toMatchSnapshot();
        });

        it('Should render with props', async () => {
            element.subheadings = ['subtitle 1', 'subtitle 2'];
            element.badges = [{text: 'Finished', icon: 'yo-home'}, {text: 'In-Progress'}];
            element.imgSrc = 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg';
            await window.flush();
            expect(element).toMatchSnapshot();
        });

        it('Should render with props and a single user', async () => {
            element.subheadings = ['subtitle 1', 'subtitle 2'];
            element.badges = [{text: 'Finished', icon: 'yo-home'}, {text: 'In-Progress'}];
            element.imgSrc = 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg';
            element.users = [{imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}];
            await window.flush();
            expect(element).toMatchSnapshot();
        });

        describe('Should render multiple users properly', () => {
            it('Should render with props and 2 users', async () => {
                element.subheadings = ['subtitle 1', 'subtitle 2'];
                element.badges = [{text: 'Finished', icon: 'yo-home'}, {text: 'In-Progress'}];
                element.imgSrc = 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg';
                element.users = [{imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}, {imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}];
                await window.flush();
                expect(element).toMatchSnapshot();
            });
            it('Should render with props and 3 users', async () => {
                element.subheadings = ['subtitle 1', 'subtitle 2'];
                element.badges = [{text: 'Finished', icon: 'yo-home'}, {text: 'In-Progress'}];
                element.users = [{imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}, {imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}, {imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}];
                element.imgSrc = 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg';
                await window.flush();
                expect(element).toMatchSnapshot();
            });
            it('Should render with props and multiple users', async () => {
                element.subheadings = ['subtitle 1', 'subtitle 2'];
                element.badges = [{text: 'Finished', icon: 'yo-home'}, {text: 'In-Progress'}];
                element.users = [{imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}, {imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}, {imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}, {imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}, {imageData: 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg'}];
                element.imgSrc = 'https://pbs.twimg.com/profile_images/960887020666150913/psDP3pv9_400x400.jpg';
                await window.flush();
                expect(element).toMatchSnapshot();
            });
        });
    });

    describe('Events', () => {

        let emitEvent, window;
        beforeEach( async () => {
            window = new TestWindow();
            emitEvent = jest.fn();
        });
    });
});