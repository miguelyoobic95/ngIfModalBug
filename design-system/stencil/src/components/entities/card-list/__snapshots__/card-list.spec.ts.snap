// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListComponent Rendering Should render 1`] = `
<yoo-card-list
  class="hydrated"
  data-yoo-card-list-host=""
>
  
  <div
    attr-layout="row"
    class="outer-container"
    data-yoo-card-list=""
  >
    
    <div
      attr-layout="column"
      class="content-container"
      data-yoo-card-list=""
    >
      <div
        class="top-container"
        data-yoo-card-list=""
      >
        <div
          attr-layout="row"
          class="heading-container"
          data-yoo-card-list=""
        >
          
        </div>
        
      </div>
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card-list=""
        data-yoo-card-list-slot=""
      >
        
        <div
          attr-layout="column"
          class="inner-container"
          data-yoo-card-list=""
        >
          
        </div>
      </div>
    </div>
  </div>
</yoo-card-list>
`;

exports[`CardListComponent Rendering Should render multiple users properly Should render with props and 2 users 1`] = `
<yoo-card-list
  class="hydrated"
  data-yoo-card-list-host=""
>
  
  <div
    attr-layout="row"
    class="outer-container"
    data-yoo-card-list=""
  >
    
    <div
      attr-layout="column"
      class="content-container"
      data-yoo-card-list=""
    >
      <div
        class="top-container"
        data-yoo-card-list=""
      >
        <div
          attr-layout="row"
          class="heading-container"
          data-yoo-card-list=""
        >
          
        </div>
        
      </div>
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card-list=""
        data-yoo-card-list-slot=""
      >
        
        <div
          attr-layout="column"
          class="inner-container"
          data-yoo-card-list=""
        >
          
        </div>
      </div>
    </div>
  </div>
</yoo-card-list>
`;

exports[`CardListComponent Rendering Should render multiple users properly Should render with props and 3 users 1`] = `
<yoo-card-list
  class="hydrated"
  data-yoo-card-list-host=""
>
  
  <div
    attr-layout="row"
    class="outer-container"
    data-yoo-card-list=""
  >
    
    <div
      attr-layout="column"
      class="content-container"
      data-yoo-card-list=""
    >
      <div
        class="top-container"
        data-yoo-card-list=""
      >
        <div
          attr-layout="row"
          class="heading-container"
          data-yoo-card-list=""
        >
          
        </div>
        
      </div>
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card-list=""
        data-yoo-card-list-slot=""
      >
        
        <div
          attr-layout="column"
          class="inner-container"
          data-yoo-card-list=""
        >
          
        </div>
      </div>
    </div>
  </div>
</yoo-card-list>
`;

exports[`CardListComponent Rendering Should render multiple users properly Should render with props and multiple users 1`] = `
<yoo-card-list
  class="hydrated"
  data-yoo-card-list-host=""
>
  
  <div
    attr-layout="row"
    class="outer-container"
    data-yoo-card-list=""
  >
    
    <div
      attr-layout="column"
      class="content-container"
      data-yoo-card-list=""
    >
      <div
        class="top-container"
        data-yoo-card-list=""
      >
        <div
          attr-layout="row"
          class="heading-container"
          data-yoo-card-list=""
        >
          
        </div>
        
      </div>
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card-list=""
        data-yoo-card-list-slot=""
      >
        
        <div
          attr-layout="column"
          class="inner-container"
          data-yoo-card-list=""
        >
          
        </div>
      </div>
    </div>
  </div>
</yoo-card-list>
`;

exports[`CardListComponent Rendering Should render with props 1`] = `
<yoo-card-list
  class="hydrated"
  data-yoo-card-list-host=""
>
  
  <div
    attr-layout="row"
    class="outer-container"
    data-yoo-card-list=""
  >
    
    <div
      attr-layout="column"
      class="content-container"
      data-yoo-card-list=""
    >
      <div
        class="top-container"
        data-yoo-card-list=""
      >
        <div
          attr-layout="row"
          class="heading-container"
          data-yoo-card-list=""
        >
          
        </div>
        
      </div>
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card-list=""
        data-yoo-card-list-slot=""
      >
        
        <div
          attr-layout="column"
          class="inner-container"
          data-yoo-card-list=""
        >
          
        </div>
      </div>
    </div>
  </div>
</yoo-card-list>
`;

exports[`CardListComponent Rendering Should render with props and a single user 1`] = `
<yoo-card-list
  class="hydrated"
  data-yoo-card-list-host=""
>
  
  <div
    attr-layout="row"
    class="outer-container"
    data-yoo-card-list=""
  >
    
    <div
      attr-layout="column"
      class="content-container"
      data-yoo-card-list=""
    >
      <div
        class="top-container"
        data-yoo-card-list=""
      >
        <div
          attr-layout="row"
          class="heading-container"
          data-yoo-card-list=""
        >
          
        </div>
        
      </div>
      <div
        attr-layout="row"
        class="slot-container"
        data-yoo-card-list=""
        data-yoo-card-list-slot=""
      >
        
        <div
          attr-layout="column"
          class="inner-container"
          data-yoo-card-list=""
        >
          
        </div>
      </div>
    </div>
  </div>
</yoo-card-list>
`;
